### PR TITLE
Mission control view plugin: test basis and two test cases

### PR DESCRIPTION
**Pull request by engelmi**
Implemented a basis for tests of the mission control plugin as well as two test cases.

### CODE REVIEW

File: src/main/java/org/jenkinsci/test/acceptance/plugins/mission_control/MissionControlView.java [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+import org.openqa.selenium.WebElement;
+
+import java.net.URL;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * {@link View} class of the mission control plugin.
+ *
+ * @author engelmi
```
Comment (uhafner) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/mission_control/MissionControlView.java]: If you want the credits put your real name here ;-)

File: src/main/java/org/jenkinsci/test/acceptance/plugins/mission_control/MissionControlView.java [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+     */
+    public void setHeightRatio(String ratioOption) {
+        ensureConfigPage();
+        heightRatio.set(ratioOption);
+    }
+
+    /**
+     * Reloads the configuration of jenkins from disk.
+     * Necessary to display the build history if a new {@link MissionControlView} is created.
+     * (Inefficient) alternative: {@code jenkins.restart()}
```
Comment (uhafner) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/mission_control/MissionControlView.java]: Is this a bug of the plug-in? Or why is this required?

File: src/test/java/plugins/MissionControlTest.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        // add shell build step to cause the job to fail
+        job.addShellStep("sh <");
+        job.save();
+        job.startBuild().waitUntilFinished();
+
+        // open mission control view and assert that no build entries are being displayed
+        // (after the creation of a new mission control view the configuration needs to be reloaded)
+        view.open();
+        WebElement buildHistory = driver.findElement(By.id("jenkinsBuildHistory"));
+        Assert.assertEquals(0, buildHistory.findElements(By.xpath(".//tbody/tr")).size(), 0);
```
Comment (uhafner) [inline src/test/java/plugins/MissionControlTest.java]: Use Harmcrest matcher
assertThat(buildHistory.findElements(By.xpath(".//tbody/tr")), hasSize(0));

File: src/test/java/plugins/MissionControlTest.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        job.configure();
+        // add shell build step to cause the job to fail
+        job.addShellStep("sh <");
+        job.save();
+        job.startBuild().waitUntilFinished();
+
+        // open mission control view and assert that no build entries are being displayed
+        // (after the creation of a new mission control view the configuration needs to be reloaded)
+        view.open();
+        WebElement buildHistory = driver.findElement(By.id("jenkinsBuildHistory"));
```
Comment (uhafner) [inline src/test/java/plugins/MissionControlTest.java]: Couldn't this be a method of the view so we can call `view.getHistory()`.

File: src/test/java/plugins/MissionControlTest.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        Assert.assertEquals(0, buildHistory.findElements(By.xpath(".//tbody/tr")).size(), 0);
+
+        // reload configuration (alternative: jenkins.restart(), very inefficient for this task)
+        view.reloadConfiguration();
+
+        // open mission control view again and assert that the n-builds are being displayed
+        view.open();
+        buildHistory = driver.findElement(By.id("jenkinsBuildHistory"));
+        Assert.assertEquals(historySize, buildHistory.findElements(By.xpath(".//tbody/tr")).size(), 0);
```
Comment (uhafner) [inline src/test/java/plugins/MissionControlTest.java]: Harmcrest

File: src/test/java/plugins/MissionControlTest.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        // check for correct highlighting of the builds
+        WebElement failedBuild = buildHistory.findElement(By.xpath(".//tbody/tr[td='" + strFailedJob + "']"));
```
Comment (uhafner) [inline src/test/java/plugins/MissionControlTest.java]: If you plan to write more history tests then this code should be placed into a PageArea sub-class.

File: src/test/java/plugins/MissionControlTest.java [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        jobFailed.configure();
+        // add shell build step to cause the job to fail
+        jobFailed.addShellStep("sh <");
+        jobFailed.save();
+        jobFailed.startBuild().waitUntilFinished();
+
+        view.open();
+        // check for the correct highlighting of the jobs
+        WebElement jobContainer = driver.findElement(By.id("jenkinsJobStatuses"));
+        WebElement btnNotBuild = jobContainer.findElement(By.xpath("//button[text()='" + strJobNotBuild + "']"));
```
Comment (uhafner) [inline src/test/java/plugins/MissionControlTest.java]: Extract into method with parameter `name`

File: src/main/java/org/jenkinsci/test/acceptance/plugins/mission_control/MissionControlView.java (hunk 1 above)
Comment (Larcagur) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/mission_control/MissionControlView.java]: Default at my pc, didn't heed it actually ;)

File: src/main/java/org/jenkinsci/test/acceptance/plugins/mission_control/MissionControlView.java (hunk 2 above)
Comment (Larcagur) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/mission_control/MissionControlView.java]: It is required (see: [https://wiki.jenkins-ci.org/display/JENKINS/Mission+Control+Plugin](url) at "Setting Up"). I don't know how mission control works internally to display the build history, but it seems the plugin developers are aware of this behavior.

File: src/test/java/plugins/MissionControlTest.java (hunk 6 above)
Comment (Larcagur) [inline src/test/java/plugins/MissionControlTest.java]: Good point. I don't know how many tests of the build history will be written, but a PageArea seems to be a better approach. Should the View provide a method to create the PageArea object (as it is only valid if the view exists) or should the PageArea be decoupled from the View?
